### PR TITLE
add keyword analyzer docs

### PR DESCRIPTION
**Pull request by AntonEliatra**
### Description
add keyword analyzer docs

### Issues Resolved
Part of https://github.com/opensearch-project/documentation-website/issues/1483 addressed in this PR.

### Version
all

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by vagimeli** [comment]
@udabhas @varun-lodaya Please provide tech review approval to move this PR forward in the documentation process. Please review this week or provide a peer who can review it. Thank you.

**Review by udabhas — APPROVED** [review]
Looks good to me!

File: _analyzers/keyword.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+layout: default
+title: Keyword analyzer
+nav_order: 80
+---
+
+# Keyword analyzer
+
+The `keyword` analyzer doesn’t tokenize the text at all. Instead, it treats the entire input as a single token. This is useful when you want the entire content of a field to be indexed as is, without breaking it into smaller pieces (tokens). The `keyword` analyzer is often used for fields like email addresses, URLs, product IDs, and other cases where tokenization is not desirable.
```
Comment (natebower) [inline _analyzers/keyword.md:9]:
```suggestion
The `keyword` analyzer doesn't tokenize text at all. Instead, it treats the entire input as a single token and does not break it into individual tokens. The `keyword` analyzer is often used for fields containing email addresses, URLs, or product IDs and in other cases where tokenization is not desirable. 
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @AntonEliatra Editorial review. Thanks!

**Review by kolchfa-aws — APPROVED** [review]

**Pull request merged**